apps/cmp.c

```diff
@@ -2293,7 +2293,7 @@ static int get_opts(int argc, char **argv)
         switch (o) {
         case OPT_EOF:
         case OPT_ERR:
-            goto opt_err;
+            goto opthelp;
         case OPT_HELP:
             opt_help(cmp_options);
             return -1;
```

Comment:
It does look a bit curious to have `OPT_ERR` jump to `opthelp`, directly followed by an `OPT_HELP`...  that doesn't.

Reply:
Yes.  It took untangling the previous goto/goto stuff to make that obvious :)  I put the opthelp body inside the switch, just like the other commands.  Curiosity solved :)